tools/ci_build/build.py

```diff
@@ -1546,7 +1546,10 @@ def build_protoc_for_host(cmake_path, source_dir, build_dir, args):
             cmd_args += ['-T', 'host=x64']
     elif is_macOS():
         if args.use_xcode:
-            cmd_args += ['-G', 'Xcode']
+            cmd_args += [
+                '-G', 'Xcode',
+                '-DCMAKE_OSX_ARCHITECTURES=x86_64'
+            ]
```

Comment:
Isn't Apple switching to ARM based chips in the relatively near future and this hardcoded default may become invalid on one of those machines? 

Could we instead check the current architecture using python and set it correctly?

Or update the MacOS build instructions so the user provides this explicitly via --cmake_extra_defines?

Reply:
Added DCMAKE_OSX_ARCHITECTURES for x86_64 Mac only

---
In reply to: [492441186](https://github.com/microsoft/onnxruntime/pull/5246#discussion_r492441186) [](ancestors = 492441186)